src/psyclone/dynamo0p3.py

```diff
@@ -308,25 +283,35 @@ def qr_basis_alloc_args(first_dim, basis_fn):
     :return: list of dimensions to use to allocate array
     :rtype: list of strings
     '''
+    if basis_fn["shape"] not in VALID_QUADRATURE_SHAPES:
```

Comment:
parent docstring should include exception description too.

Reply:
Done.